Core/Src/main.c

```diff
@@ -87,19 +95,27 @@ int main(void)
   /* Initialize all configured peripherals */
   MX_GPIO_Init();
   MX_USART2_UART_Init();
+  MX_SPI1_Init();
+  MX_TIM1_Init();
   /* USER CODE BEGIN 2 */
-
+  HAL_TIM_PWM_Start(&htim1, TIM_CHANNEL_1);
   /* USER CODE END 2 */
 
```

Comment:
Good practice to pull the CS line high before communication starts (exactly what you do on line 111), otherwise the first message or first few messages can be lost. Sometimes this causes code to crash.